classes/session/SessionManager.php

```diff
@@ -234,25 +237,13 @@ public function gc(int $lifetime): bool
      */
     public function updateSessionCookie($sessionId = false, int $expireTime = 0): bool
     {
-        $domain = ini_get('session.cookie_domain');
-        // Specific domains must contain at least one '.' (e.g. Chrome)
-        if (strpos($domain, '.') === false) {
-            $domain = false;
-        }
-
-        // Clear cookies with no domain #8921
-        if ($domain) {
-            setcookie(session_name(), '', 0, ini_get('session.cookie_path'), false);
```

Comment:
This `Set-Cookie` will be overridden by the `Set-Cookie` that follows below (cookie names are unique at the browser, so isn't not needed to clear "old cookies").
This is also responsible for the unexpected issues at Cypress.

Reply:
...and 8921 is here: http://wayback.archive-it.org/7100/20180509190857/https://pkp.sfu.ca/bugzilla/show_bug.cgi?id=8921

Reply:
@asmecher I tried to find this issue in GitHub and failed, thanks for the link!

The discussion makes sense, but the problem still exists (Chrome + OJS 3.3.0):
1. Accessed `journal.xyz`
   > Got a cookie with the domain `.journal.xyz`.
2. Accessed `www.journal.xyz`
   > The previous cookie is visible at the subdomain, and no extra cookie was created.
3. Tried to login and got redirected back to the login form
   > A new cookie with the domain `.www.journal.xyz` was created, but the previous cookie wasn't removed, because:
    - The code above only drops domainless cookies
    - The previous cookie failed to be re-used, because this line doesn't work (it's not possible to set these session constants when there's an active session): https://github.com/pkp/pkp-lib/blob/ef79d952408a9c91fa0b38cfbdb407df69429704/classes/session/SessionManager.php#L78

FYI, the login form generated these `Set-Cookie` headers at the response:
```http
Set-Cookie: OJSSID=mlamfo2l6deevkhnjmflsmk1tk; path=/; domain=www.journal.xyz
Set-Cookie: OJSSID=deleted; expires=Thu, 01-Jan-1970 00:00:01 GMT; Max-Age=0; path=/
Set-Cookie: OJSSID=mlamfo2l6deevkhnjmflsmk1tk; path=/; domain=www.journal.xyz; HttpOnly
Set-Cookie: OJSSID=deleted; expires=Thu, 01-Jan-1970 00:00:01 GMT; Max-Age=0; path=/
Set-Cookie: OJSSID=mlamfo2l6deevkhnjmflsmk1tk; expires=Thu, 09-Feb-2023 19:43:34 GMT; Max-Age=2592000; path=/; domain=www.journal.xyz; HttpOnly
```

I'll check the other bugzilla link and update the PR.